src/db/file.rs

```diff
@@ -356,4 +385,28 @@ mod test {
         let files = get_file_list(env::current_dir().unwrap().join("Cargo.toml")).unwrap();
         assert_eq!(files[0], std::path::Path::new("Cargo.toml"));
     }
+    #[test]
+    fn test_mime_types() {
+        check_mime("/ignored", ".gitignore", "text/plain");
+        check_mime("[package]", "hello.toml","text/x-toml");
+        check_mime(".ok { color:red; }", "hello.css","text/css");
+        check_mime("var x = 1", "hello.js","application/javascript");
+        check_mime("<html>", "hello.html","text/html");
+        check_mime("## HELLO", "hello.hello.md","text/markdown");
+        check_mime("## WORLD", "hello.markdown","text/markdown");
+        check_mime("{}", "hello.json","application/json");
+        check_mime("hello world", "hello.txt","text/plain");
```

Comment:
Please add a test case for rust source code. In particular, [we depend](https://github.com/rust-lang/docs.rs/blob/master/src/web/source.rs#L132) on rust source starting with `text/`.